src/string_buffer.rs

```diff
@@ -198,6 +204,41 @@ impl fmt::Display for StringBuffer {
     }
 }
 
+impl fmt::Debug for StringBuffer {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "\"{}\"", self)
+    }
+}
+
+impl PartialEq for StringBuffer {
+    fn eq(&self, other: &StringBuffer) -> bool {
+        // Shortcut if sizes differ
+        if self.len != other.len { return false }
+
+        {
+            let mut this_last = &*self.first;
+            let mut other_last = &*other.first;
+
+            loop {
+                if this_last.data != other_last.data { return false }
+                let this_next = this_last.next.as_ref();
+                let other_next = other_last.next.as_ref();
+
+                match (this_next, other_next) {
+                    (Some(this_next), Some(other_next)) => {
+                        this_last = this_last;
+                        other_last = other_next;
+                    },
+                    (None, None) => return true,
+                    _ => return false
+                }
+            }
+        }
+    }
```

Comment:
This will return false in the case that two StringBuffers have the same content but a different structure, e.g., `["Hello world"] vs ["Hello ", "world"]`. I think we should return true in that case, although I'm not sure how best to do that without allocating new StringBuffers